esmvalcore/cmor/table.py

```diff
@@ -988,51 +988,52 @@ class CustomInfo(CMIP5Info):
 
     Parameters
     ----------
-    cmor_tables_path: str or None
+    cmor_tables_path:
         Full path to the table or name for the table if it is present in
-        ESMValTool repository
-    """
-
-    def __init__(self, cmor_tables_path=None):
-        cwd = os.path.dirname(os.path.realpath(__file__))
-        default_cmor_folder = os.path.join(cwd, 'tables', 'custom')
+        ESMValTool repository. If ``None``, use default tables from
+        `esmvalcore/cmor/tables/custom`.
 
-        # Get custom location of CMOR tables if possible
-        if cmor_tables_path is None:
-            self._cmor_folder = default_cmor_folder
-        else:
-            self._cmor_folder = self._get_cmor_path(cmor_tables_path)
-        if not os.path.isdir(self._cmor_folder):
-            raise ValueError(f"Custom CMOR tables path {self._cmor_folder} is "
-                             f"not a directory")
+    """
 
+    def __init__(self, cmor_tables_path: Optional[str | Path] = None) -> None:
+        """Initialize class member."""
+        self.coords = {}
         self.tables = {}
-        self.var_to_freq = {}
+        self.var_to_freq: dict[str, dict] = {}
         table = TableInfo()
         table.name = 'custom'
         self.tables[table.name] = table
 
-        # Try to read coordinates from custom location, use default location if
-        # not possible
-        coordinates_file = os.path.join(
-            self._cmor_folder,
-            'CMOR_coordinates.dat',
-        )
-        if os.path.isfile(coordinates_file):
-            self._coordinates_file = coordinates_file
+        # First, read default custom tables from repository
+        self._cmor_folder = self._get_cmor_path('custom')
+        self._read_table_dir(self._cmor_folder)
+
+        # Second, if given, update default tables with user-defined custom
+        # tables
+        if cmor_tables_path is not None:
+            self._custom_cmor_folder = self._get_cmor_path(cmor_tables_path)
+            if not os.path.isdir(self._custom_cmor_folder):
+                raise ValueError(
+                    f"Custom CMOR tables path {self._custom_cmor_folder} is "
+                    f"not a directory"
+                )
+            self._read_table_dir(self._custom_cmor_folder)
```

Comment:
this is slightly confusing variable naming (am not pointing fingers at you, since these were already in, but maybe a good chance to change them?) default CMOR stock: cmor tables, custom tables inside ESMValCore: custom tables, user specified tables: user tables IMHO

Reply:
prob best addressed in a separate PR BTW

Reply:
Agreed! I changed the new `_custom_cmor_folder` to `_user_table_dir` in https://github.com/ESMValGroup/ESMValCore/pull/2279/commits/feed776791ad0bce8e08c2bd44048e8908409167, but did not change `_cmor_folder` for consistency with the other tables 👍